### PR TITLE
Bootstrap v4.1

### LINKED ISSUE #71: update to bootstrap v4.1
Hey @stefanneculai 👋 

[Bootstrap](http://getbootstrap.com/) released their final `v4.0.0` back in January. 💜 
Would like to contribute some work I've [started](https://github.com/m5o/design-blocks/tree/bootstrap-v4.1) with `4.0` and now `4.1`. 
Hope you're interested in this update and some more cleanup around that.

📝 https://blog.getbootstrap.com/2017/12/28/bootstrap-4-beta-3/
📝 https://blog.getbootstrap.com/2018/01/18/bootstrap-4/
📝 https://blog.getbootstrap.com/2018/04/09/bootstrap-4-1/

### PR DESCRIPTION
**Pull request by m5o**
### update to bootstrap `v4.1` 💜 

📺 [Demo Preview](http://froala-design-blocks-bs4.bitballoon.com)

I've intentionally did a lot of small commits, so you could follow my upgrade path: 
* update bootstrap css and js references to `v4.0.0`
* replace input-group-btn with input-group-append
* update year to 2018
* update fontawesome to `v5.0.8`
* use font awesome `fa-angle-right`
* use bootstrap spacing utilities
* re-use bootstrap color variables
* add semicolon for consistency
* use `fa-sign-in-alt`
* use `small.text-muted`
* update dependencies in `package.json`
* structure froala scss resources like bootstrap scss
* correct border attributes
* simplify teams stylesheets
* use `theme-color` map
* setup autoprefixer
* fix indentation
* separate header and footer styles
* convert inline navigator styles to scss
* move inline fdb-block styles to scss
* use bootstrap `border-radius` utilities
* remove unused `br-b-0` class
* use bootstrap background utilities
* use bootstrap dark utilities
* refactor custom `no-border` class
* remove unused `.text-light` style
* prefix heart with `text-`
* remove unused footer-small style
* remove unused `.fdb-icon-round`
* use bootstrap navbar color schemes
* fix pricings issue with dark background
* add text utilities with reset and popular brands
* refactor footer with text-reset and text brand helper
* remove unused custom mixins
* move text-heart to text utils
* add slack to brand colors
* re-use bootstrap button component
* use bootstrap cards
* refactor fdb-image-bg
* rename fdb-touch to card-touch
* refactor custom fill-left / fill-right
* define bootstrap spacer utilities
* use bootstrap `.font-weight-light`
* rename to `footer-lg`
* use bootstrap typography classes
* convert `px` to `rem`
* add bg-image helper
* change sass output style to expanded
* add opacity helper
* use margin utilities for images
* add icons to pricings table
* re-use bootstrap media object
* re-use bootstrap forms layout and styles
* validate html pages
* update path in readme
* fix minor alignment issues with customers logos
* use bootstrap form markup similar to forms.html
* remove strong markup from headlines
* minor markup and style optimizations
* use font-awesome brands for some examples
* improve froala navigator for small viewports
* add text util for xbrand
* npm update bootstrap
* use bootstrap shadow utils
* minor pricings improvements
* remove unused custom box-shadow mixin
* use shadow util directly without btn-shadow proxy
* update cdn libraries to `v4.1.0`
* update font awesome to `v5.0.9`
* use opacity util

🐞  Resolves #71 

📝 https://blog.getbootstrap.com/2017/12/28/bootstrap-4-beta-3/
📝 https://blog.getbootstrap.com/2018/01/18/bootstrap-4/
📝 https://blog.getbootstrap.com/2018/04/09/bootstrap-4-1/

### CODE REVIEW

**Comment by stefanneculai** [comment]
@m5o thanks for the PR. Amazing work! Most of the changes look great, however, there are a few things we'd want to keep as they are now. We'll review the changes carefully and get back with detailed feedback.

**Comment by m5o** [comment]
@stefanneculai Friendly ping 🛎  Do you already can add some feedback?

**Comment by stefanneculai** [comment]
Could you ping us via our website contact form? Would love to chat a little bit about them. Thanks in advance!

File: src/scss/froala_blocks.scss [hunk 1]
```diff
@@ -1,246 +1,23 @@
-@import './buttons';
```
Comment (stefanneculai) [inline src/scss/froala_blocks.scss]: We should be using our own variables and mixins instead of the Boostrap default ones.

File: src/scss/fullwidth.scss [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.fullwidth-split {
```
Comment (stefanneculai) [inline src/scss/fullwidth.scss]: It looks like this is adding new classes over the existing ones, which we don't want to do in a small realease.

Comment (m5o) [reply src/scss/fullwidth.scss]: What do you think about shipping a major release instead? 
There were a lot of changes, small and huge between Bootstrap `v4.0.0-beta.2` and `v4.1`. Please read all the blog posts and focus on the 'Breaking changes' and 'What's new' sections.

📝 https://blog.getbootstrap.com/2017/12/28/bootstrap-4-beta-3/
📝 https://blog.getbootstrap.com/2018/01/18/bootstrap-4/
📝 https://blog.getbootstrap.com/2018/04/09/bootstrap-4-1/

Comment (stefanneculai) [reply src/scss/fullwidth.scss]: Give us a couple of days to think about shipping a major version. We should get with a response early next week regarding a major release.

File: src/scss/froala_blocks.scss (hunk 1 above)
Comment (m5o) [reply src/scss/froala_blocks.scss]: 🤔 I suggest a direction, where re-useing bootstrap button variables instead of custom btn overrides. The refactoring is clear and isolated here: 3709cc621eff29c7ab70808314f0b3c6b1e10051

Your project should follow variables introduces by the bootstrap origin.

**Comment by m5o** [comment]
Well, it seems your requests for changes are hard to follow for an upgrade this size. Bootstrap beta changed a lot, this is why it took them so long. You have to accept this fact to move this project further. ✌️🌻

**Comment by stefanneculai** [comment]
We would be fine to make a `1.1.0` version, however, we don't want to change anything for now in the overall design of the blocks - the PR includes changes to the button styles and to the way a design block look. If you could adapt it in such a way that the design of the blocks do not change, we would be absolutely happy to merge it.

**Comment by m5o** [comment]
Happy to her that a `v1.1.0` is accepted 🚀

When updating, revising and applying the latest bootstrap changes in this mega release, it is practically impossible not to visually change the appearance of any of the blocks. I may not understand your request correctly. Perhaps you can explain it to me using an example? (Screenshot with annotations e.g.) 🤷‍♂️

Regarding the buttons, Bootstrap got [Outline Buttons](http://getbootstrap.com/docs/4.1/components/buttons/#outline-buttons). A welcome addition, therefore using the buttons from the native library instead of your selfmade alternative. Shouldn't be so far apart.

**Comment by stefanneculai** [comment]
Sure, please see below few screenshots. I understand that the temptation of making changes is high, however, the layout of the blocks that we currently have has been carefully designed and we reviewed lots of common cases for it.

screen shot 2018-07-16 at 18 54 19
screen shot 2018-07-16 at 18 52 23

**Comment by m5o** [comment]
@stefanneculai Do you have the generated froala docs links by hand? Is the preview page still online? I couldn't find a link.

**Comment by stefanneculai** [comment]
Run `npm run gulp` and then open http://localhost:8001 in the browser.

**Comment by m5o** [comment]
| before   | after  |
|---|---|
|  before  |  
block
 |

I could change the buttton to use `btn-primary` if this is ok?

Just btw. before the button had only a single `.btn` class, this is not a valid bootstrap button class. Bootstrap requires the button style class witch is a second class then.

```html
<a href="https://www.froala.com" class="btn">Call to Action</a>
```

**Comment by stefanneculai** [comment]
That sounds good - there are other places where we have the buttons these way, so I think it would be good to do it everywhere.. 

I think we could go through all the design blocks in the upcoming days and point out all such blocks.

Best,

Stefan

**Comment by m5o** [comment]
| before   | after  |
|---|---|
|  before  | screen_shot_2018-07-17_at_18_18_36 |

I could change…:
* sub-headline to `.h2`
* `.lead` 📝  https://getbootstrap.com/docs/4.1/content/typography/#lead
* the buttton to use `btn-primary`

**Comment by stefanneculai** [comment]
We'd prefer to keep the overall design the same and not change the way buttons, texts and images are. We could, of course, use Bootstrap buttons such as `btn-primary` instead of the `btn` as we currently have - that would be a huge improvement.

Do you use Slack? We could talk faster on https://froala-design-blocks-slack.herokuapp.com/.

**Comment by m5o** [comment]
> We'd prefer to keep the overall design the same and not change the way buttons, texts and images are.

Thank you for communicating that open. I guess we're having different directions then, that are harder to sync than I thought before. Beside from the suggestion to use `btn-primary` I have made also some small corrections to copytext and for example color of the icons, which then change the "overall design". We have August now and I don't see us coming near to a finish line.

**Comment by m5o** [comment]
@stefanneculai any word from your side? Do you care?

**Comment by aaronhudon** [comment]
@m5o I'll be following and contributing to your bootstrap 4.1 fork.

**Comment by stefanneculai** [comment]
@m5o we've discussed this and we're willing to accept the changes required by Bootstrap 4.1, however, we don't want to make any changes to the overall design of the blocks (buttons, font size, backgrounds, etc.). We'd be happy to merge this as soon as the changes are made this way.
